shpc/utils/fileio.py

```diff
@@ -61,6 +61,28 @@ def mkdir_p(path):
             logger.exit("Error creating path %s, exiting." % path)
 
 
+def rmdir_to_base(path, base_path):
+    """
+    Delete the tree under $path and all the parents
+    up to $base_path as long as they are empty
+    """
+    if not os.path.isdir(base_path):
+        logger.exit("Error: %s is not a directory" % base_path)
+    if not path.startswith(base_path):
+        logger.exit("Error: %s is not a parent of %s" % (base_path, path))
+
+    if os.path.exists(path):
```

Comment:
Any reason to have this delete up here? Won't it happen in the while loop?

Reply:
The delete in the for loop is conditioned to the directory being empty, but at this stage `path` is unlikely to be empty. It will likely contain a container, or a module, or a `bin/`, etc